lib/components/tutorial.js

```diff
@@ -0,0 +1,241 @@
+      };
+
+      return _react2.default.createElement(
+        _stepThrough2.default,
+        { ref: function ref(el) {
+            _this2.stepThrough = el;
+          }, className: 'tutorial-steps', style: tutorialStyle },
+        this.props.tutorial.steps.map(function (step, i) {
+          if (!step._key) {
+            step._key = Math.random();
+          };
+          var source = void 0;
+          var button = void 0;
+
+          if (_this2.props.media[step.media]) {
+            source = _this2.props.media[step.media].src;
+          };
+          if (i === _this2.props.tutorial.steps.length - 1) {
+            button = _react2.default.createElement(
+              'button',
+              { type: 'submit', className: 'major-button' },
+              'Let\u2019s go!'
+            );
+          } else {
+            button = _react2.default.createElement(
+              'button',
+              { type: 'button', className: 'standard-button', onClick: _this2.handleNextClick },
+              'Continue'
+            );
+          }
+
+          return _react2.default.createElement(
+            _mediaCard2.default,
+            { key: step._key, className: 'tutorial-step', src: source },
+            _react2.default.createElement(
+              _markdownz.Markdown,
+              null,
+              step.content
+            ),
+            _react2.default.createElement('hr', null),
+            _react2.default.createElement(
+              'p',
+              { style: { textAlign: 'center' } },
+              button
+            )
+          );
+        })
+      );
+    }
+  }]);
+
+  return Tutorial;
+}(_react2.default.Component);
+
+;
+
+Tutorial.checkIfCompleted = function (tutorial, user, preferences) {
+  if (user) {
+    window.prefs = preferences;
+    if (preferences && preferences.preferences && preferences.preferences.tutorials_completed_at && preferences.preferences.tutorials_completed_at[tutorial.id]) {
+      return Promise.resolve(preferences.preferences.tutorials_completed_at[tutorial.id]);
+    }
+  } else {
+    if (completedThisSession[tutorial.id]) {
+      return Promise.resolve(completedThisSession[tutorial.id]);
+    }
+  }
+};
+
+Tutorial.startIfNecessary = function (self, tutorial, user, preferences) {
+  if (tutorial) {
+    self.checkIfCompleted(tutorial, user, preferences).then(function (completed) {
+      if (!completed) {
+        self.start(tutorial, user, preferences);
+      }
+    });
+  }
+};
+
+Tutorial.find = function (workflow) {
+  if (workflow) {
+    _apiClient2.default.type('tutorials').get({ workflow_id: workflow.id }).then(function (tutorials) {
+      var onlyStandardTutorials = tutorials.filter(function (tutorial) {
+        return tutorial.kind === 'tutorial' || null;
+      });
+      return onlyStandardTutorials[0];
+    });
+  } else {
+    Promise.resolve();
+  }
+};
+
+Tutorial.start = function (TutorialComponent, tutorial, user, preferences) {
```

Comment:
`TutorialComponent` is an example of where the binding seems a bit odd in this component. You can see where this is being used in PFE [here](https://github.com/zooniverse/Panoptes-Front-End/blob/0f3b9c02c4b8e031bc5e4fa8d3674ea170c9281f/app/classifier/classifier.jsx#L295). I think it's a bit odd to have static methods in the [tutorial component](https://github.com/zooniverse/Panoptes-Front-End/blob/0f3b9c02c4b8e031bc5e4fa8d3674ea170c9281f/app/components/tutorial.cjsx) that refer to `this`, but perhaps there was a reason for this?

Reply:
Static methods should be pure functions, in my opinion, so I think you're right to explicitly pass `TutorialComponent` as the first argument. 

The PFE code makes `start` look like an instance method, which can't be right because the component instance is created when `start` runs.

Reply:
Yeah, it's a bit odd how the Tutorial component operates in PFE.